### ISSUE TITLE
[Phase 1] Data Infrastructure - HDF5 Loading

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/gravitational-wave-detection#2]
## Objective

Build the data loading infrastructure for gravitational wave strain data from HDF5 files.

**Part of:** #1

## Dependencies

None (first phase)

### Data Loader Implementation
- [ ] Create `src/data/loader.py` with `GWDataLoader` class
- [ ] Support G2Net HDF5 format
- [ ] Load all three detectors (H1, L1, V1)

### Preprocessor
- [ ] Create `src/data/preprocessor.py` with `SignalPreprocessor`
- [ ] Implement bandpass filtering (20-500 Hz)
- [ ] Add whitening capability

### Configuration
- [ ] Create `configs/datasets.yaml` with dataset metadata

### Tests
- [ ] Write unit tests in `tests/unit/test_loader.py`
- [ ] Verify data shapes and types

### GWDataLoader Skeleton

```python
# src/data/loader.py
import numpy as np
import h5py
from typing import Optional, Tuple

DATASET_CONFIG = {
    "g2net": {
        "sample_rate": 2048,
        "duration": 2.0,
        "detectors": ["H1", "L1", "V1"],
    }
}

class GWDataLoader:
    def __init__(
        self,
        dataset: str = "g2net",
        data_dir: str = "data/raw",
        sample_size: Optional[int] = None,
        random_state: int = 42
    ):
        self.dataset = dataset
        self.data_dir = data_dir
        self.sample_size = sample_size
        self.random_state = random_state

    def load(self) -> Tuple[np.ndarray, np.ndarray]:
        """Load strain data and labels.

        Returns:
            X: shape (n_samples, 3, 4096) - 3 detectors, 2s at 2048Hz
            y: shape (n_samples,) - binary labels
        """
        pass

    def load_split(self, test_size=0.2):
        """Load and split into train/test."""
        pass
```

### First Test

```python
# tests/unit/test_loader.py
import pytest
from src.data.loader import GWDataLoader

def test_loader_config():
    loader = GWDataLoader("g2net")
    assert loader.config["sample_rate"] == 2048
    assert len(loader.config["detectors"]) == 3
```

## Definition of Done

- [ ] Can load G2Net HDF5 files
- [ ] Each sample has shape (3, 4096)
- [ ] Labels are binary (0 or 1)
- [ ] Unit tests pass

## Technical Notes

- G2Net files organized in subdirectories by ID prefix
- Each HDF5 contains H1, L1, V1 datasets
- 2 seconds of data at 2048 Hz = 4096 samples
